orchprovider/k8s/v1/k8s.go

```diff
@@ -1051,19 +1055,58 @@ func (k *k8sOrchestrator) createReplicaDeployment(volProProfile volProfile.Volum
 
 	glog.Infof("Adding replica(s) for Volume '%s'", vsm)
 
+	repLabelSpec := map[string]string {
+		string(v1.VSMSelectorKey):               vsm,
+		string(v1.PVCSelectorKey):               pvc,
+		string(v1.VolumeProvisionerSelectorKey): string(v1.JivaVolumeProvisionerSelectorValue),
+		string(v1.ReplicaSelectorKey):           string(v1.JivaReplicaSelectorValue),
+		// -- if manual replica addition
+		//string(v1.ReplicaCountSelectorKey):      strconv.Itoa(rCount),
+	}
+
+	//Add the application label to the replica deployment if it exists.
+	appLV := vol.Labels.ApplicationOld
+	if appLV != "" {
+		repLabelSpec[string(v1.ApplicationSelectorKey)] = appLV
+	}
+
+	//Set the Default Replica Topology Key ( kubernetes.io/hostname )
+	replicaTopoKey :=  v1.GetPVPReplicaTopologyKey(nil)
+
+	//One of the labels to match will always be a constant, which is 
+	// specific to OpenEBS. This is to avoid collision with application pods.
+	repAntiAffinityLabelSpec := map[string]string {
+		string(v1.ReplicaSelectorKey): string(v1.JivaReplicaSelectorValue),
+	}
+
+	//Check if a custom topology key has been provided for this volume.
+	// Note: The custom topology keys have to be passed via the PVCs as labels.
+	// And since these labels don't allow special characters like '/' in the value,
+	// the topology key has been divided into domain and type. 
+	// Examples:
+	//   kubernetes.io/hostname
+	//   failure-domain.beta.kubernetes.io/zone
+	//   failure-domain.beta.kubernetes.io/region
+	replicaTopoKeyDomainLV := vol.Labels.ReplicaTopologyKeyDomainOld
+	replicaTopoKeyTypeLV := vol.Labels.ReplicaTopologyKeyTypeOld
+
+	//Depending on the topology key, additional label selectors may be required. 
+	if replicaTopoKeyDomainLV  != "" && replicaTopoKeyTypeLV  != "" {
+		replicaTopoKey =  replicaTopoKeyDomainLV   + "/" + replicaTopoKeyTypeLV  
+		//TODO : We are assuming here that the topology keys depend on
+		// the application label.
+		repAntiAffinityLabelSpec[string(v1.ApplicationSelectorKey)] = appLV
```

Comment:
We are already set `repAntiAffinityLabelSpec` above ?

Reply:
yes. we initialized it with a common value that is required in all cases. Updating here with either applicationkey or vsmkey, based on the condition.